ab_tool/management/commands/unassigned_students.py

```diff
@@ -1,32 +1,38 @@
+import logging
+import traceback
 from django.core.management.base import BaseCommand
 
 from ab_tool.canvas import get_unassigned_students_with_stored_credentials
 from ab_tool.exceptions import NoValidCredentials
 from ab_tool.models import Experiment
+from ab_tool.constants import (ASSIGN_STUDENTS_MESSAGE, NO_CREDENTIALS_MESSAGE)
+from ab_tool.controllers import send_email_notification
+
+
+logger = logging.getLogger(__name__)
+
 
 class Command(BaseCommand):
     args = ""
     help = "Sends email alerts about courses with unassigned students"
     
     def handle(self, *args, **options):
-        # TODO: update notification messages
         # TODO: Include intent-to-treat experiments when that is supported
         for experiment in Experiment.get_all_started_csv():
-            course_notification = experiment.get_course_notification()
-            # Check whether or not a notification can be sent to avoid extra API calls
-            if not course_notification.can_notify():
-                continue
             try:
-                students = get_unassigned_students_with_stored_credentials(course_notification, experiment)
-                if students:
-                    self.send_email_notification(course_notification, "unnasigned students")
-            except NoValidCredentials:
-                self.send_email_notification(course_notification, "no valid credentials")
+                self.handle_experiment(experiment)
+            except Exception as exception:
+                logger.error(repr(exception))
+                logger.error(traceback.format_exc())
     
-    def send_email_notification(self, course, message):
-        # Potentially redundant notification check in case this method is used
-        # in other places in the future
-        if course.can_notify():
-            self.stdout.write(str(course.get_emails()))
-            self.stdout.write("TODO: send email %s" % message)
-            course.notification_sent()
+    def handle_experiment(self, experiment):
+        course_notification = experiment.get_course_notification()
+        # Check whether or not a notification can be sent to avoid extra API calls
+        if not course_notification.can_notify():
```

Comment:
I think it would read better if you nested the try/catch under the if condition above, but without the not.